api/src/main/java/org/geysermc/geyser/api/entity/EntityData.java

```diff
@@ -49,6 +49,16 @@ public interface EntityData {
      */
     @NonNull CompletableFuture<@Nullable GeyserEntity> entityByJavaId(@NonNegative int javaId);
 
+    /**
+     * Returns a {@link GeyserEntity} to e.g. update entity properties.
+     */
+    @NonNull CompletableFuture<@Nullable GeyserEntity> entityByUuid(@NonNegative UUID javaUuid);
```

Comment:
These don't need to be CompletableFutures because get checks on Fastutil maps can be concurrent. Modifying them cannot be done off-thread. Not sure how to deal with an API break like the one above - it was absolutely a terrible decision to make early on because it's simply unneeded.